src/Nethermind/Nethermind.Ssz/Crypto/Ssz.BlsPublicKey.cs

```diff
@@ -0,0 +1,49 @@
+//  Copyright (c) 2018 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.Runtime.CompilerServices;
+using Nethermind.Core2.Crypto;
+
+namespace Nethermind.Ssz
+{
+    public static partial class Ssz
+    {
+        public static void Encode(Span<byte> span, BlsPublicKey value)
```

Comment:
I am not biggest fan of those static methods instead of some kind of polymorphism maybe? But I might be wrong.

Reply:
it is not case for generics if you think about it. generics can be used where you have a template that behaves exactly the same no matter the type. here we have a different behavioir for each type

Reply:
I split into classes in case later we can just build generics for decoders - here yes - we could resolve a generic decoder with Decode, Encode classes

Reply:
(split into files)

Reply:
we have this generics approach in RLP so maybe will use same here

Reply:
yes, considering it - after I know more how they are going to be used I may end up with a generics style approach with Encode / Decode like in Rlp

Reply:
but there may be some significant performance overhead - the latest commit slowed down the thing by 40% (but made it much more readable), very possible that for now I do not need to think about optimizations more as this might be totally outside of the hot paths

Reply:
I think this may evolve a lot like RLP did